src/api/resources/base.py

```diff
@@ -347,9 +355,12 @@ def is_valid_field(self, model, field_name):
         except FieldDoesNotExist:
             return False
 
-    def remove_invalid_fields(self, queryset, ordering, view, request):
-        return [term for term in ordering
-                if self.is_valid_field(queryset.model, term.lstrip('-'))]
+    def remove_invalid_fields(self, queryset, fields, view, request):
+        valid_fields = [item[0] for item in self.get_valid_fields(queryset, view, {'request': request})]
+        valid_model_fields = [term for term in fields
+                              if self.is_valid_field(queryset.model, term.lstrip('-'))]
+        valid_fields = set(valid_fields + valid_model_fields)
```

Comment:
Do you need to maintain the order of fields with `remove_invalid_fields`? If so you won't be able to use `set`